[zacklow28] iP

zacklow28 opened the pull request:
### FeedMe

- easy to use like `code`
    - cause it is 
    - only difference is you get to specify a file first
        - if file is found, load that file
        - if file not found, create a new file with specified name 

1. download it [here](https://github.com/zacklow28/ip)
2. use it ✅ 

- [x] Done
- [ ] Not Done 

This is the code for $\frac {greet} {meet}$
```Java
public class Main {
    public static void main(String[] args) {
        ui.greet();    
    }
}
```
> FeedMe more!

*Free* $\cup$ **Free** $\cap$ ~Free~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
level0

Commit:
level0

Commit:
level0

Commit:
level0

Commit:
level0

Commit:
level1

Commit:
level2

Commit:
level4

Commit:
Bug Fix

Commit:
Update FeedMe.java

Commit:
testing

Commit:
level5

Commit:
added delete function

Commit:
Junit test

Commit:
added memory feature

Commit:
Merge branch 'branch-Level-7'

Commit:
added date type to deadline and event

Commit:
Merge branch 'branch-Level-8'

Commit:
added Storage and Tasklist

Commit:
added Ui and Parser

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
added packages

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge branch 'add-gradle-support'

Commit:
added gradle, checkstyle edits

Commit:
added JUnit tests, cleanup some formatting

Commit:
Merge branch 'branch-A-JUnit'

Commit:
build/libs JAR

Commit:
Merge branch 'branch-A-JAR'

Commit:
added JavaDocs

Commit:
Remove deleted file

Commit:
Edit to follow coding standard

Commit:
Add find task function

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Accept both input from merge conflict

Commit:
Follow checkstyle guide and improve naming

Commit:
Merge branch 'branch-A-CheckStyle'

feliciahmq commented on src/main/java/feedme/Parser.java:
```diff
@@ -0,0 +1,80 @@
+     */
+    public static String parse(String in) {
+        String[] inArr = in.split(" ", 2); //split just on the first space
+        if (in.equals("bye")) {
```
Perhaps can use switch case instead of if else here. We can use ENUMS too.

feliciahmq commented on src/main/java/feedme/Storage.java:
```diff
@@ -0,0 +1,136 @@
+                }
+                break;
+            case 'D':
+                String[] foodArr = food.split(" by: ");
```
Maybe we can parse the line and create a new ToDo/ Deadline/ Event inside their own subclass instead of here?

feliciahmq commented on src/main/java/feedme/Ui.java:
```diff
@@ -0,0 +1,155 @@
+     * @param br buffer reader for reading user input
+     * @throws IOException if input cannot be read
+     */
+    public void respondToUserBasedOnCommand(String in, Tasklist tasklist, Storage storage, BufferedReader br)
```
Why not create specific response functions for each kind of command and call it in their command subclass? eg. in todo, call showToDo()

feliciahmq commented on src/main/java/feedme/Parser.java:
```diff
@@ -0,0 +1,80 @@
+        String[] inArr = in.split(" ", 2); //split just on the first space
+        if (in.equals("bye")) {
+            return "goodbye";
+        } else if (in.equals("list")) {
```
Perhaps can extract out Command general class, and have specific subclasses?

feliciahmq commented on src/test/java/feedme/task/TasklistTest.java:
```diff
@@ -0,0 +1,41 @@
+
+import org.junit.jupiter.api.Test;
+
+public class TasklistTest {
```
I like how thorough the tests are.

feliciahmq reviewed (commented):
LGTM 👍🏻

Commit:
Fix names to code standard

Commit:
Add dependencies and pictures

Commit:
Break up method to better fit GUI implementation

Commit:
Add javafx stage 2

Commit:
Set up javafx stage 3

Commit:
Set up javafx stage 4

Commit:
Add FXML files

Commit:
Fix bug on mouse click and text enter coordination

Commit:
Merge branch 'branch-Level-10'

Commit:
Add vararg feature to find method

Commit:
Merge branch 'branch-A-Varargs'

amir-dha commented on src/main/java/feedme/MainWindow.java:
```diff
@@ -0,0 +1,124 @@
+     */
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
```
Perhaps, you could introduce line break in these types of statements to enhance readability.

amir-dha commented on src/main/java/feedme/Main.java:
```diff
@@ -0,0 +1,30 @@
+package feedme;
```
Perhaps, you could break it into more specific packaging.

amir-dha commented on src/test/java/feedme/ParserTest.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test
+    public void parseInputToCommandTask() {
```
Perhaps variable names could be more succinct? I noted this in other places too.

amir-dha reviewed (commented):
This looks really good. The style and the code are comprehensive and very readable.

Commit:
Add JAR file and delete redundant file

Created the updated Fat JAR file that incorporates JavaFX libaries,
allowing the user to interact with the chat GUI.

Removed the redundant file FeedMe as now the main class is Main and
called using the Launcher class. FeedMe is no longer used.

Commit:
Add assertions

Use assert feature to document important assumptions that should hold in the code. For method in Parser, assert that the task type is correct before use. For first edited method in Ui, assert that array size is more than 0 before continuing with function. For second edited method in Ui, assert that task list should have a size more than zero before executing the code.

These assertions ensure that the programmer's assumptions are correct and stop the run if there are any discrepancies

Commit:
Improve code quality

Extracted long method into shorter functions. Resolved some complicated
expressions. Improved some magic strings and values by turning them into
constants.

Commit:
Delete dead code

zacklow28 committed:
Merge pull request #1 from zacklow28/branch-A-Assertions

Add assertions

zacklow28 committed:
Merge branch 'master' into branch-A-CodeQuality

zacklow28 committed:
Merge pull request #2 from zacklow28/branch-A-CodeQuality

Branch A-CodeQuality

Commit:
Fix small wording error

Commit:
Add Update function to task list

Commit:
Merge branch 'branch-C-Update'

Commit:
Fix minor JavaDoc issue

Commit:
Add resizing feature

Commit:
Add user guide

Commit:
Merge branch 'branch-A-UserGuide'

Commit:
Fix Ui picture link

Commit:
Add some important details to README

Commit:
Update README

Commit:
Add Styling

Commit:
Add simple comment

Commit:
Style dialog box

zacklow28 committed:
Merge pull request #5 from zacklow28/Styling

Style dialog box
